src/Domain/FunctionsOfTime/FunctionOfTimeHelpers.cpp

```diff
@@ -68,10 +68,10 @@ const StoredInfo<MaxDerivPlusOne, StoreCoefs>& stored_info_from_upper_bound(
          "Vector of StoredInfos you are trying to access is empty. Was it "
          "constructed properly?");
 
-  const auto upper_bound_stored_info = std::upper_bound(
+  const auto upper_bound_stored_info = std::lower_bound(
       all_stored_infos.begin(), all_stored_infos.end(), t,
```

Comment:
If we are using `std::lower_bound` now, should we change the name of the function? It might be a little confusing to have the function named `stored_info_from_upper_bound` when it actually uses `lower_bound`

Reply:
Describe the interface, not the implementation.  The function takes an upper bound on the times it wants, not a lower bound.

Reply:
Ok.